### DISCUSSION TITLE
How do I iterate on attributes by category?

### DISCUSSION THREAD

**Discussion by Oreolek** [evennia/evennia#3092] [Evennia Questions] [answered]
I wrote this:

```python
        attrs = self.attributes.get(category='rules', return_obj=True)
        for attr in attrs:
            try
                rules[attr.key] = attr.value
            except Error:
                # do nothing if attr has no key or value
```
and I don't know if this is right or wrong way to do it

**Comment by InspectorCaracal** [comment, marked as answer]
It's _a_ way to do it, although I'm left wondering what you want to do this _for._ What's the purpose of loading the category of attributes into a dictionary? If you want your `rules` in dictionary format, it might be simpler to store the whole dictionary in a single attribute.

Also, I believe you can safely assume that all Attributes gotten this way are going to have a value for `.key` and `.value` will return something, so the try/except block shouldn't be necessary.

**Comment by Oreolek** [comment]
The single attribute would be long and won't fit into one line.

**Reply by Griatch** [reply]
Your version works, not sure why storing it as one dict would necessarily be longer though: 
 
```python
rules = obj.db.rules   # where rules is a dict
for rule_name, rule_value in rules.items(): 
     # ... 

```

You don't need to do this loop just to check anyway, especially if you won't do anything if it doesn't exist. With a dict, you can just do `rules.get("rulename")` and check that the result exists before doing anything with it.
